### PR TITLE
add listeners for pingSent & pongReceived

### PR DESCRIPTION
**Pull request by faizan-carameltech**
I needed to make some decisions on basis of number of failed pings and pong timeouts. Therefore publicly handling listeners for these events.

### CODE REVIEW

File: Source/SocketEngineClient.swift [hunk 1]
```diff
@@ -26,6 +26,8 @@
 import Foundation
 
 @objc public protocol SocketEngineClient {    
+    func engineDidSentPing()
```
Comment (nuclearace) [inline Source/SocketEngineClient.swift]: `engineDidSendPing`

File: Source/SocketEngineClient.swift [hunk 2]
```diff
@@ -26,6 +26,8 @@
...
+    func engineDidReceivedPong()
```
Comment (nuclearace) [inline Source/SocketEngineClient.swift]: `engineDidReceivePong`

File: Source/SocketIOClient.swift [hunk 3]
```diff
@@ -275,6 +275,19 @@ public final class SocketIOClient : NSObject, SocketEngineClient, SocketParsable
         }
     }
 
+    ///ping & pong
+    public func engineDidSentPing() {
+        DefaultSocketLogger.Logger.error("%@", type: logType, args: "")
```
Comment (nuclearace) [inline Source/SocketIOClient.swift]: This should have better logging info.  This should also be log, not error.

File: Source/SocketIOClient.swift [hunk 4]
```diff
@@ -275,6 +275,19 @@ public final class SocketIOClient : NSObject, SocketEngineClient, SocketParsable
...
+        
+        handleEvent("sentPing", data: [""], isInternalMessage: true)
+    }
+    
+    public func engineDidReceivedPong() {
+        DefaultSocketLogger.Logger.error("%@", type: logType, args: "")
```
Comment (nuclearace) [inline Source/SocketIOClient.swift]: This should have better logging info. This should also be log, not error.

Comment (faizan-carameltech) [inline Source/SocketIOClient.swift]: What other info you require in logs? I will accommodate that. Furthermore closing this PR and opening a new one against development branch with the suggested changes.

**Pull request closed without merging**